server/aws-lsp-codewhisperer/src/language-server/chat/chatController.test.ts

```diff
@@ -0,0 +1,253 @@
+import * as assert from 'assert'
+import sinon from 'ts-sinon'
+import { createIterableResponse } from '../testUtils'
+import { ChatController } from './chatController'
+import { ChatSessionManagementService } from './chatSessionManagementService'
+import { ChatSessionService } from './chatSessionService'
+
+describe('ChatController', () => {
+    const mockTabId = 'tab-1'
+    const mockConversationId = 'mock-conversation-id'
+    const mockMessageId = 'mock-message-id'
+
+    const mockAssistantResponseList: ChatResponseStream[] = [
+        {
+            assistantResponseEvent: {
+                content: 'Hello ',
+            },
+        },
+        {
+            assistantResponseEvent: {
+                content: 'World',
+            },
+        },
+        {
+            assistantResponseEvent: {
+                content: '!',
+            },
+        },
+    ]
+
+    const expectedCompleteChatResult: ChatResult = {
+        messageId: mockMessageId,
+        body: 'Hello World!',
+        canBeVoted: undefined,
+        codeReference: undefined,
+        followUp: undefined,
+        relatedContent: undefined,
+    }
+
+    const mockCancellationToken = {
+        isCancellationRequested: false,
+        onCancellationRequested: () => ({ dispose: () => null }),
+    }
+
+    let generateAssistantResponseStub: sinon.SinonStub
+    let disposeStub: sinon.SinonStub
+    let clock: sinon.SinonFakeTimers
+
+    let testFeatures: TestFeatures
+    let chatSessionManagementService: ChatSessionManagementService
+    let chatController: ChatController
+
+    beforeEach(() => {
+        clock = sinon.useFakeTimers()
+        generateAssistantResponseStub = sinon
+            .stub(CodeWhispererStreaming.prototype, 'generateAssistantResponse')
+            .callsFake(() => {
+                return new Promise(resolve =>
+                    setTimeout(() => {
+                        resolve({
+                            conversationId: mockConversationId,
+                            $metadata: {
+                                requestId: mockMessageId,
+                            },
+                            generateAssistantResponseResponse: createIterableResponse(mockAssistantResponseList),
+                        })
+                    }, 100)
+                )
+            })
+
+        testFeatures = new TestFeatures()
+
+        disposeStub = sinon.stub(ChatSessionService.prototype, 'dispose')
+
+        chatSessionManagementService = ChatSessionManagementService.getInstance().withCredentialsProvider(
+            testFeatures.credentialsProvider
+        )
+
+        chatController = new ChatController(chatSessionManagementService, testFeatures)
+    })
+
+    afterEach(() => {
+        generateAssistantResponseStub.restore()
+        disposeStub.restore()
+        ChatSessionManagementService.reset()
+        clock.restore()
+    })
+
+    it('chatController creates a session when a tab add notifcation is received', () => {
```

Comment:
Won't `chatController` be duplicated in test names, because it's also in `describe`?

Reply:
Good catch thanks! I removed the duplicated `chatController` text.